psc-package/Main.hs

```diff
@@ -0,0 +1,259 @@
+{-# LANGUAGE DeriveAnyClass #-}
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE NamedFieldPuns #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE RecordWildCards #-}
+{-# LANGUAGE TupleSections #-}
+
+module Main where
+
+import qualified Data.Aeson as Aeson
+import           Data.Aeson.Encode.Pretty
+import           Data.Foldable (fold, for_, traverse_)
+import           Data.List (nub)
+import qualified Data.Map as Map
+import           Data.Maybe (mapMaybe)
+import qualified Data.Set as Set
+import           Data.Text (pack)
+import qualified Data.Text.Lazy as TL
+import qualified Data.Text.Lazy.Builder as TB
+import           Data.Text.Encoding (encodeUtf8)
+import           Data.Traversable (for)
+import           Data.Version (showVersion)
+import qualified Filesystem.Path.CurrentOS as Path
+import           GHC.Generics (Generic)
+import qualified Options.Applicative as Opts
+import qualified Paths_purescript as Paths
+import qualified System.IO as IO
+import           Turtle hiding (fold)
+
+packageFile :: Path.FilePath
+packageFile = "psc-package.json"
+
+data PackageConfig = PackageConfig
+  { name    :: Text
+  , depends :: [Text]
+  , set     :: Text
+  , source  :: Text
+  } deriving (Show, Generic, Aeson.FromJSON, Aeson.ToJSON)
+
+pathToTextUnsafe :: Turtle.FilePath -> Text
+pathToTextUnsafe = either (error "Path.toText failed") id . Path.toText
+
+defaultPackage :: Text -> PackageConfig
+defaultPackage pkgName =
+  PackageConfig { name    = pkgName
+                , depends = [ "prelude" ]
+                , set     = "psc-" <> pack (showVersion Paths.version)
```

Comment:
The reasons I'd like to keep them together:

- I'd like the default install to have some sensible packaging solution without having to use NPM or another package manager to install additional tools.
- Eventually I'd like to think about compiler integrations like checking minor version bumps are really non-breaking.
- I'd like to get to the point where we have one executable before 1.0, and then this would be run as `psc package <command>`.